src/tel_download.js

```diff
@@ -180,28 +244,52 @@
       }
     } else if (!ele.querySelector("._tel_download_button_img")) {
       // add download button to images
-      const imageUrl = ele.querySelector("img.thumbnail").src;
-
-      const container = document.createElement("div");
-      container.className = "_tel_download_button_img_container";
-      container.style.position = "absolute";
-      container.style.width = "100%";
-      container.style.height = "100%";
-      container.style.display = "flex";
-      container.style.justifyContent = "center";
-      container.style.alignItems = "end";
-      const downloadButton = document.createElement("button");
-      downloadButton.className =
-        "btn-icon default__button _tel_download_button_img";
-      downloadButton.innerHTML = downloadIcon;
-      downloadButton.style.marginBottom = "16px";
-      downloadButton.style.backgroundColor = "black";
-      downloadButton.onclick = (e) => {
-        e.stopPropagation();
-        tel_download_image(imageUrl);
-      };
-      ele.appendChild(container);
-      container.appendChild(downloadButton);
+      const image = ele.querySelector("img.thumbnail");
+      if(image != null){
+        let imageUrl = image.src;
+        const container = document.createElement("div");
+        container.className = "_tel_download_button_img_container";
+        container.style.position = "absolute";
+        container.style.width = "100%";
+        container.style.height = "100%";
+        container.style.display = "flex";
+        container.style.justifyContent = "center";
+        container.style.alignItems = "end";
+        const downloadButton = document.createElement("button");
+        downloadButton.className =
+            "btn-icon default__button _tel_download_button_img";
+        downloadButton.innerHTML = downloadIcon;
+        downloadButton.style.marginBottom = "16px";
+        downloadButton.style.backgroundColor = "black";
+        downloadButton.onclick = (e) => {
+            e.stopPropagation();
+            tel_download_image(imageUrl);
+        };
+        ele.appendChild(container);
+        container.appendChild(downloadButton);
+      }else {
+        let gifUrl = ele.querySelector("video").src;
+        const container = document.createElement("div");
```

Comment:
The code here has a lot of overlap with the above if branch. Could we potentially eliminate this redundancy?

Reply:
Thanks! It looks great to me now. I will do some testing and approve once I verify it works.